doc/Language/traits.pod6

```diff
@@ -108,6 +108,27 @@ would be a way of adding functionality which cuts across class and role
 hierarchies, or can even be used to add behaviors to independently defined
 routines.
 
+=head2 X<C<is implementation-detail>,traits> trait
+
+Note: this trait is available in Rakudo compiler starting from 2020.05 release.
+
+The C<is> with C<implementation-detail> argument is used by Raku
+language implementations and module authors to mark particular
+routines (including methods) as not meant to be a public API.
+
+Applying this trait makes the C<is-implementation-detail> method on
+L<Code|/type/Code> to return C<True>.
+
+    my &fail-routine = &fail;
+    unless &fail-routine.is-implementation-detail {
+        say "It is safe to use &fail routine in your modules!";
+    }
```

Comment:
Why the reference to safety? Shouldn't you clarify that before, implying that implementation details can be removed at any time from the API?

Reply:
The example tries to say the `fail` routine is ok to use in your modules, no? Extended the explanation, though.

Reply:
The problem is what you mean by "is ok". An implementation detail can be used as long as you test for it, for instance, or have an easy way out if it's changed. So "safe" or "is OK" are not really shorthand for that...

Reply:
Just to clarify a possible misunderstanding...
This condition tests if the routine (`fail`) is *not* an implementation detail, and then says "If it is not (and it is not), then it is safe to use it". Is it wrong?

Reply:
Not really wrong, but I'm not sure "safe" is the right word for it. It's "safe" as long as it does not change, but it might change; again, it's "safe" if you future-proof your code by testing for those features and having a B plan in case they change.

Reply:
What about "&fail is not not an implementation detail, can expect backward compatibility"?

Reply:
except for the double "not", I guess it's better.